src/TLuaInterpreter.cpp

```diff
@@ -5341,8 +5336,8 @@ int TLuaInterpreter::setPopup(lua_State* L)
         // removes value, but keeps key for next iteration
         lua_pop(L, 1);
     }
-    if (!lua_istable(L, ++s)) {
-        lua_pushstring(L, "setPopup: wrong argument type");
+    if (!lua_istable(L, s++)) {
```

Comment:
Are you sure about this - it changes the index of the stack item being tested to check that it is a table to the item whose not being a table is being reported and whose index is being reported. i.e. the item tested to see if it is a table might be the 3rd one - but it is reported as the 4th item and it is the type of the 4th item that will be reported NOT the 3rd one...

Don't forget that the item at the top of the stack is in index position **1** not **0**.

Reply:
I agree. No idea why I changed this :sweat_smile: